components/eamxx/src/dynamics/homme/eamxx_homme_iop.cpp

```diff
@@ -138,26 +138,23 @@ advance_iop_subsidence(const KT::MemberType& team,
     const auto fac = (dt/2)/pdel(k);
 
     // Update u
-    auto& u_k = u(k);
-    u_k.set(at_top, u_k - fac*omega_int_kp1*delta_u_k);
-    u_k.set(at_bot, u_k - fac*omega_int_k*delta_u_km1);
-    u_k.set(at_mid, u_k - fac*(omega_int_kp1*delta_u_k + omega_int_k*delta_u_km1));
+    u(k).update(at_top, fac*omega_int_kp1*delta_u_k, -1, 1);
```

Comment:
Maybe we can be a bit faster here (and below) and only do two updates? Like

```suggestion
auto not_top = not at_top;
auto not_bot = not at_bot;
u(k).update (not_at_top, fac*omega_int_k*delta_u_km1, -1, 1);
u(k).update (not_at_bot, fac*omega_int_kp1*delta_u_k, -1, 1);
```

Though it be not bfb...

Reply:
Good call. BFB is not important.